In [1]:
 # Google Colab Setup
from google.colab import drive

# This will prompt for authorization.
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [0]:
import os

In [0]:
os.chdir('/content/drive/My Drive/Analytic-Steps-blogs/YouTube/')

In [0]:
import pandas as pd
import numpy as np

In [5]:
data = pd.read_csv("ExtractedData.csv")
data.head()

title  ... comment
0  Saavi credits NPTEL courses for helping her in...  ...       2
1  One course can change one's life - it did chan...  ...       1
2  From coding courses to coding rounds - how Pay...  ...       3
3  NPTEL courses taught Mohit the skill of attent...  ...       0
4  Harischandra's interest to explore the finer n...  ...       0

[5 rows x 8 columns]

In [6]:
print("total number of videos uploaded on the channel: "+str(data.shape[0]))

total number of videos uploaded on the channel: 20060


In [12]:
data['views'].value_counts()

439      15
759      13
472      12
445      12
1040     11
         ..
71083     1
1449      1
42044     1
56738     1
4170      1
Name: views, Length: 12389, dtype: int64

In [9]:
data['likes'].value_counts()

2       1092
1       1057
3       1039
4        936
0        753
        ... 
683        1
539        1
507        1
342        1
3662       1
Name: likes, Length: 734, dtype: int64

In [10]:
data['comment'].value_counts()
data['dislikes'].value_counts()

0      8480
1      3327
2      1869
3      1173
4       856
       ... 
104       1
136       1
168       1
265       1
199       1
Name: comment, Length: 147, dtype: int64

In [7]:
data['dislikes'].value_counts()

0      9022
1      3487
2      1945
3      1142
4       819
       ... 
74        1
90        1
106       1
122       1
183       1
Name: dislikes, Length: 117, dtype: int64

In [140]:
data['likes'].isnull().value_counts()

False    20060
Name: likes, dtype: int64

In [141]:
data.videoIDS.isnull().value_counts()

False    20060
Name: videoIDS, dtype: int64

In [142]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20060 entries, 0 to 20059
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   title              20060 non-null  object
 1   videoIDS           20060 non-null  object
 2   video_description  19287 non-null  object
 3   publishedDate      20060 non-null  object
 4   likes              20060 non-null  int64 
 5   dislikes           20060 non-null  int64 
 6   views              20060 non-null  int64 
 7   comment            20060 non-null  int64 
dtypes: int64(4), object(4)
memory usage: 1.2+ MB


In [143]:
data.describe()

likes      dislikes         views       comment
count  20060.000000  20060.000000  2.006000e+04  20060.000000
mean      59.291775      3.352742  1.788379e+04      4.528913
std      181.454593      9.393482  5.212895e+04     14.929778
min        0.000000      0.000000  1.500000e+01      0.000000
25%        5.000000      0.000000  1.521000e+03      0.000000
50%       15.000000      1.000000  4.531500e+03      1.000000
75%       49.000000      3.000000  1.512775e+04      4.000000
max     7069.000000    381.000000  1.882922e+06    717.000000

In [144]:
print("Maxium number of likes on videos are ",data['likes'].max())
print("Maxium number of dislikes on videos are ",data['dislikes'].max())
print("Maxium number of view on videos are ",data['views'].max())
print("Maxium number of comments on videos are ",data['comment'].max())

Maxium number of likes on videos are  7069
Maxium number of dislikes on videos are  381
Maxium number of view on videos are  1882922
Maxium number of comments on videos are  717


## Sort the videos on the basis of likes i.e. most liked video will appear at last

In [0]:
data_liked = data.sort_values(by = 'likes',axis = 0,ascending = True,ignore_index=True)

In [148]:
data.head()

title  ... comment
0  Saavi credits NPTEL courses for helping her in...  ...       2
1  One course can change one's life - it did chan...  ...       1
2  From coding courses to coding rounds - how Pay...  ...       3
3  NPTEL courses taught Mohit the skill of attent...  ...       0
4  Harischandra's interest to explore the finer n...  ...       0

[5 rows x 8 columns]

In [149]:
data_liked.head()

title     videoIDS  ... views comment
0   Mod-01 Lec-33 Lecture-33  HZC7oTy93x4  ...   208       0
1           CAM00039 Student  L7MkESmQWig  ...   267       0
2     Mod-01 Lec-37 Test - 2  jKbwtkyA1Y8  ...   661       0
3  CAM00031 RNB Engg College  uTJS2QxVYnY  ...    97       0
4     CAM00033 Shyamsankaran  Iz3em-m70Cg  ...   282       0

[5 rows x 8 columns]

we will now find out the count of unique values of videos i.e. count of likes,dislikes etc. using **value_counts**

In [16]:
data['likes'].value_counts(ascending = True,bins = 10)

(4948.3, 5655.2]        1
(5655.2, 6362.1]        1
(6362.1, 7069.0]        1
(3534.5, 4241.4]        2
(4241.4, 4948.3]        3
(2827.6, 3534.5]        5
(2120.7, 2827.6]       13
(1413.8, 2120.7]       29
(706.9, 1413.8]       136
(-7.07, 706.9]      19869
Name: likes, dtype: int64

In [15]:
data['dislikes'].value_counts(ascending= True, bins = 10)

(266.7, 304.8]        0
(304.8, 342.9]        0
(228.6, 266.7]        1
(342.9, 381.0]        1
(152.4, 190.5]        4
(190.5, 228.6]        4
(114.3, 152.4]       12
(76.2, 114.3]        37
(38.1, 76.2]        131
(-0.382, 38.1]    19870
Name: dislikes, dtype: int64

as from the above cell, it is clear that most of the videos have likes between 0 to 1767 i.e. total 20022 <br>
and the most liked videos are less. Also if go more deeper then it will be clear that only one video have highest likes

In [151]:
data_liked['dislikes'].value_counts()

0      9022
1      3487
2      1945
3      1142
4       819
       ... 
74        1
90        1
106       1
122       1
183       1
Name: dislikes, Length: 117, dtype: int64

From the above, it is clear that only 3 videos are most disliked

In [152]:
data['comment'].max()

717

In [153]:
data['comment'].value_counts(ascending = True)

199       1
265       1
168       1
136       1
104       1
       ... 
4       856
3      1173
2      1869
1      3327
0      8480
Name: comment, Length: 147, dtype: int64

From the above cell it can be concluded that there is only one video with maxium comments i.e. 199 comments. <br>
apart from this count of videos with no comments are maximum i.e. 8480 videos

## Now we will extract the details of the most liked videos,most comment videos, videos with highest view

In [0]:
most_likedID = data['likes'].idxmax()
most_viewdID = data['views'].idxmax()
most_commentID = data['comment'].idxmax()

In [155]:
data.iloc[most_likedID].head

<bound method NDFrame.head of title                Lecture - 1 Introduction to Quantum Physics;He...
videoIDS                                                   TcmGYe39XG0
video_description    Lecture Series on Quantum Physics by Prof.V.Ba...
publishedDate                                     2008-12-17T06:52:25Z
likes                                                             7069
dislikes                                                           220
views                                                          1118460
comment                                                            717
Name: 17009, dtype: object>

In [156]:
data.iloc[most_viewdID]

title                    Lecture - 1 Introduction to Basic Electronics
videoIDS                                                   w8Dq8blTmSA
video_description    Lecture Series on Basic Electronics by Prof. T...
publishedDate                                     2008-01-22T06:29:53Z
likes                                                             5467
dislikes                                                           381
views                                                          1882922
comment                                                            523
Name: 19729, dtype: object

In [157]:
data.iloc[most_commentID]

title                Lecture - 1 Introduction to Quantum Physics;He...
videoIDS                                                   TcmGYe39XG0
video_description    Lecture Series on Quantum Physics by Prof.V.Ba...
publishedDate                                     2008-12-17T06:52:25Z
likes                                                             7069
dislikes                                                           220
views                                                          1118460
comment                                                            717
Name: 17009, dtype: object